### PR TITLE
[Test] Fix fusion-group mocks after runtime context migration

### PR DESCRIPTION
**Pull request by BBuf**
## Motivation

After #40068 migrated FlashInfer communication-group reads to the runtime context, `TestResolveFusionGroup` still patches `get_tp_group`, `get_moe_ep_group`, and `get_moe_tp_group` in `flashinfer_comm_fusion`. Those imports no longer exist, so all three group-resolution cases fail with `AttributeError` while entering the mock context. This also blocks the Base CPU suite ([failure log](https://github.com/sgl-project/sglang/actions/runs/35418955183/job/105832992563)).

## Modifications

Supply the three fake group handles through the existing `get_parallel().override(...)` context and remove the obsolete patches. Only the test fixture changes; all existing topology cases and assertions are retained.

## Accuracy Tests

Reproduced on unmodified main `111b905bd1eec22329dd10a8e5a5d9fbc2f9753c`, then reran with this test-only patch in the same environment:

```bash
PYTHONPATH=python python test/registered/unit/layers/test_layer_communicator_fusion_gate.py
```

- Before: 18 tests, 3 errors (missing `get_tp_group`).
- After: **18 tests passed**.
- Applicable pre-commit hooks and `git diff --check` passed.

These CPU logic tests do not initialize a distributed process group. Model accuracy evaluation is not applicable.

The separate main-branch Ruff formatting failure in `logits_processor.py` is addressed in #40265 and is outside this test-only fix.

## Speed Tests and Profiling

Not applicable: no inference code changes.

## Checklist

- [x] Code formatted with pre-commit.
- [x] Existing regression tests reproduced and validated.
- [x] Follows SGLang code style.
- Documentation and model accuracy/speed benchmarks: not applicable.

---
### CI States

Latest PR Test (Base): :x: [Run #35426884578](https://github.com/sgl-project/sglang/actions/runs/35426884578)
Latest PR Test (Extra): :x: [Run #35426884389](https://github.com/sgl-project/sglang/actions/runs/35426884389)
Latest PR Test (AMD ROCm 10): :x: [Run #35426884627](https://github.com/sgl-project/sglang/actions/runs/35426884627)

### CODE REVIEW

**Comment by BBuf** [comment]
/rerun-test test/registered/unit/layers/test_layer_communicator_fusion_gate.py

**Comment by BBuf** [comment]
/rerun-test test/registered/unit/layers/test_layer_communicator_fusion_gate.py
